lib.php

```diff
@@ -123,16 +123,43 @@ public function get_links($linkarray) {
         if (array_key_exists('error', $results)) {
             return $results['error'];
         }
+
+        global $DB;
+        $plagiarismvalues = $DB->get_records_menu('plagiarism_vericite_config', array('cm' => $vericite['cmid']), '', 'name,value');
```

Comment:
This is terribly inefficient. This function gets calls for every single submission, so in a class of 100's of students, you'd be looking up the same record 100's of times. Is there a different place to put this logic or get this setting?